apps/fabric-website/src/components/Nav/Nav.tsx

```diff
@@ -96,34 +96,45 @@ export class Nav extends React.Component<INavProps, INavState> {
   private _renderLinkList(pages: INavPage[], isSubMenu: boolean, title?: string): React.ReactElement<{}> {
     const { filterState } = this.state;
 
-    const links = pages.filter(page => !page.hasOwnProperty('isHiddenFromMainNav')).map((page: INavPage, linkIndex: number) => {
-      if (page.isCategory && !filterState) {
-        return <span>{page.pages.map((innerPage: INavPage, innerLinkIndex) => this._renderLink(innerPage, innerLinkIndex))}</span>;
-      }
-      return page.isCategory && filterState ? this._renderCategory(page, linkIndex) : this._renderLink(page, linkIndex);
-    });
+    const links = pages
+      .filter(page => {
+        // Don't render pages that are explicitly told not to, as well as top-level pages that aren't active.
+        if (page.hasOwnProperty('isHiddenFromMainNav') || (page.isUhfLink && !_hasActiveChild(page) && !_isPageActive(page))) {
```

Comment:
can be simplified to 

```ts
.filter(page => 
   !page.hasOwnProperty('isHiddenFromMainNav') && !(page.isUhfLink && !_hasActiveChild(page) || 
   _isPageActive(page))
.map(...
```

Reply:
That's fair. I personally find one-liners like this a little unintuitive to scan, but I suppose that's what the comment is for.